docs/source/basic/cheatsheet.rst

```diff
@@ -256,6 +256,12 @@ Atomic operations
          AtomicAdd, AtomicSub, AtomicMin, AtomicMax, AtomicExch,
          AtomicInc, AtomicDec, AtomicAnd, AtomicOr, AtomicXor, AtomicCas
 
+Memory fences on block- or device level (guarantees LoadLoad and StoreStore ordering)
+  .. code-block:: c++
+
+     mem_fence(acc, memory_scope::Block{});
+     mem_fence(acc, memory_scope::Device{});
```

Comment:
Can we not reuse the same hierarchy tags as for the atomics?

Reply:
We could, but I was afraid this would cause confusion. We can only fence within a block (which mirrors `ThreadAtomic`) or the whole device (`GridAtomic`), but not between blocks of the same grid (`BlockAtomic`).

Reply:
Yeah I see, better have a separate hierarchy!